### PR TITLE
Tle writeclusterrep

### PR DESCRIPTION
**Pull request by tuanle618**
Code to check: https://pastebin.com/DRmQ0Sym

Currently the width of the plots for the combination cluster-plots is somehow not properly. I tried the settings in rmarkdown with fig.width to several numbers but still the result was not satisfying. As of now it works without errors. But in default mode, when automatically selecting numeric columns it might happen that clustering algorithm does not converge.

### CODE REVIEW

File: R/getClusterAnalysis.R [hunk 1]
```diff
@@ -207,7 +220,7 @@ getClusterAnalysis = function(data, num.features, method, par.vals, random.seed,
       proc.data = as.data.frame(cbind(pca.data$x, cluster = kernel.cluster@.Data))
       proc.data$cluster = as.factor(proc.data$cluster)
       ##ggscatter
-      kernel.plot = ggscatter(data = proc.data, x = "PC1", y = "PC2",
+      kernel.plot = ggpubr::ggscatter(data = proc.data, x = "PC1", y = "PC2",
```
Comment (MiGraber) [inline R/getClusterAnalysis.R:223]: why did you add the `::` operator?
You should use instead:
`@importFrom ggpubr ggscatter`

**Comment by MiGraber** [comment]
Pastebin L 37-end:
r code is fine with no errors, but when rendering html:
```4: Computation failed in stat_ellipse():
der führende Minor der Ordnung 2 ist nicht positiv definit
5: Computation failed in stat_ellipse():
der führende Minor der Ordnung 2 ist nicht positiv definit
6: Computation failed in stat_ellipse():
der führende Minor der Ordnung 1 ist nicht positiv definit
```
Nevertheless the file renders properly (except for the plots you mentioned)

File: R/getClusterAnalysis.R (hunk 1 above)
Comment (ptl93) [reply R/getClusterAnalysis.R]: I get an error when importing packe ggpubr or functiion ggscatter into the namespace. Instead I put it in DESCRIPTION

Comment (tuanle618) [reply R/getClusterAnalysis.R:223]: I get an error when importing package ggpubr or function ggscatter from ggpubr into the namespace. Instead I put it in DESCRIPTION

**Comment by MiGraber** [comment]
First par of the pastebin example: (I used a smaller subsample of the iris data set)
Should the pcr plot look like this
grafik
The combination plots also look wrong:
grafik
The Clustering seems to be wrong. Do you know if this is caused by the method "dbscan"?
The legend for the first graph is fine but for the second (and the following ones) the color and shape are not rendered together.

For kkmeans:
grafik
The axis labels are all the same for all comb. plots

For the render process of knitr I get the same errors as above mentioned

**Comment by tuanle618** [comment]
Yes, I believe DBScan by default generates this plot when applying the algorithm on the principal component data as well as for the combinations. I will have a look into the axis and legends. 

Normally dbscan will be applied to numeric data, which have some kind of patterns like "o-Shape" or somehing like this. For the legends I believe it shoule be right, since I only called the default method of dbscan. Its probably the algorithm leads with its default parameters to the result.

Have a look at: http://www.sthda.com/english/articles/30-advanced-clustering/105-dbscan-density-based-clustering-essentials/

I'll fix the axis problem soon.
